### PR TITLE
Allow polymorphism of TheVoice component

### PR DESCRIPTION
**Pull request by mcellier**
## Description

Allow polymorphism for TheVoice component. Will help seach & seo pages to use normalize components with the right margins. 

## What has been done

- Stopped using <Title> for this component
- Use typography component for text

## Things to consider

In theory we might be able to use `forwardedAs` from `styled-components` but i didn't succeed. Any idea?

## How it was tested

Storybook

### CODE REVIEW

File: src/theVoice/TheVoice.tsx [hunk 1]
```diff
@@ -1,21 +1,17 @@
...
 
 import { StyledTheVoice } from './TheVoice.style'
 
 export type TheVoiceProps = Readonly<{
+  children: ReactNode
   id?: string
   className?: string
-  children: ReactNode
   isInverted?: boolean
+  as?: keyof JSX.IntrinsicElements | React.ComponentType<any> // from styled-components types
```
Comment (guillaumewuip) [inline src/theVoice/TheVoice.tsx:10]: Maybe can we only allow some tags like `h1 | h2 | h3` ?

Comment (mcellier) [reply src/theVoice/TheVoice.tsx:10]: We're using a default styled-component feature here. We might render it as p, span or div, it doesn't matter IMHO.

Comment (Tuxosaurus) [reply src/theVoice/TheVoice.tsx:10]: IMO if we're gonna add "as" to a component then it should behave like the other instances with that prop, so no restriction I guess.

**Review by Tuxosaurus — APPROVED** [review]
Thanks also for the id on Text, I'm guessing we'll need that soon™

File: src/theVoice/TheVoice.tsx (hunk 1 above)
Comment (guillaumewuip) [reply src/theVoice/TheVoice.tsx:10]: Ok. I'm more in favour of providing only what is needed right now and edit it for new usage latter. Smallest API first for an easy maintenance and robustness in the long term.

For example are we sure we want to allow `span` or `div` here? And even if we want to allow them, is it not more explicit to have them listed in the type like `| 'span' // you can use span here for <reason>` ?

That's not blocking :)

**Pull request merged**